src/xdp/programinspect.c

```diff
@@ -406,6 +458,8 @@ XdpParseFrame(
     Cache->UdpCached = TRUE;
     Cache->TcpCached = TRUE;
     Cache->TransportPayloadCached = TRUE;
+    Cache->Icmp4Cached = TRUE;
```

Comment:
Subjectively, I don't think we should proactively parse and cache ICMP headers here. Instead, we should do what QUIC does, which is parse the inner header only if a QUIC rule gets executed.

Reply:
The reasoning: XDP is a performance feature, and we don't want every single packet to have to validate headers like ICMP/ICMPv6 if common programs are only running rules like "redirect UDP traffic" or "redirect IP traffic". We want the rules engine to pay the parsing costs of packets only if an application's rule requires it.

The cut line for proactive parsing here is fairly arbitrary; we figure that most rules will want the IP header and many will want either UDP/TCP headers, so it makes sense to do those first few unconditionally and simplify the rules processing loop. We expect ICMP is much less common for XDP rules than UDP/TCP.